catalyst_voices/packages/catalyst_voices_blocs/lib/src/registration/cubits/wallet_link_cubit.dart

```diff
@@ -66,7 +74,80 @@ final class WalletLinkCubit extends Cubit<WalletLinkStateData>
   }
 
   @override
-  void submitRegistration() {
-    // TODO(dtscalac): submit RBAC transaction
+  Future<void> prepareRegistration(SeedPhrase seedPhrase) async {
+    try {
+      emit(
+        state.copyWith(
+          unsignedTx: const Optional(null),
+          submittedTx: const Optional(null),
+          isSubmittingTx: false,
+        ),
+      );
+
+      final walletApi = await state.selectedWallet!.wallet.enable();
+
+      final registrationBuilder = RegistrationTransactionBuilder(
+        // TODO(dtscalac): inject the networkId
+        transactionConfig: await txConfigRepository.fetch(NetworkId.testnet),
+        networkId: NetworkId.testnet,
+        seedPhrase: seedPhrase,
+        roles: state.selectedRoles ?? state.defaultRoles,
+        changeAddress: await walletApi.getChangeAddress(),
+        rewardAddresses: await walletApi.getRewardAddresses(),
+        utxos: await walletApi.getUtxos(
+          amount: Balance(
+            coin: CardanoWalletDetails.minAdaForRegistration,
+          ),
+        ),
+      );
+
+      final tx = await registrationBuilder.build();
+      emit(state.copyWith(unsignedTx: Optional(Success(tx))));
+    } on Exception catch (error, stackTrace) {
+      _logger.severe('prepareRegistration', error, stackTrace);
+      emit(state.copyWith(unsignedTx: Optional(Failure(error))));
+      rethrow;
```

Comment:
why do you want to `rethrow` here? it wont be catched anywhere and will crash app. Same in `submitRegistration`

Reply:
This won't be needed anymore as I'm moving this code into registration cubit but the idea was that after submitting a transaction in case of success I had to call nextStep on registration cubit. Since that code lives in WalletLinkCubit from where I don't have access to RegistrationCubit I had to call nextStep from the widget, thus needed to know if submittingTransaction worked or not. Now since I moved the code it's not needed because I can call nextStep() from RegistrationCubit directly